src/test/java/org/broadinstitute/hellbender/tools/copynumber/formats/SampleLocatableCollectionUnitTest.java

```diff
@@ -0,0 +1,199 @@
+        enum SimpleLocatableTableColumn {
+            CONTIG,
+            START,
+            END,
+            VALUE;
+
+            static final TableColumnCollection COLUMNS = new TableColumnCollection((Object[]) values());
+        }
+        
+        private static final Function<DataLine, SimpleLocatable> SIMPLE_LOCATABLE_DATA_LINE_TO_RECORD_FUNCTION = dataLine -> {
+            final String contig = dataLine.get(SimpleLocatableTableColumn.CONTIG);
+            final int start = dataLine.getInt(SimpleLocatableTableColumn.START);
+            final int end = dataLine.getInt(SimpleLocatableTableColumn.END);
+            final int value = dataLine.getInt(SimpleLocatableTableColumn.VALUE);
+            final SimpleInterval interval = new SimpleInterval(contig, start, end);
+            return new SimpleLocatable(interval, value);
+        };
+
+        private static final BiConsumer<SimpleLocatable, DataLine> SIMPLE_LOCATABLE_RECORD_AND_DATA_LINE_BI_CONSUMER = (simpleLocatable, dataLine) ->
+                dataLine.append(simpleLocatable.getInterval().getContig())
+                        .append(simpleLocatable.getInterval().getStart())
+                        .append(simpleLocatable.getInterval().getEnd())
+                        .append(simpleLocatable.getValue());
+
+        private SimpleSampleLocatableCollection(final File inputFile) {
+            super(inputFile, SimpleLocatableTableColumn.COLUMNS, SIMPLE_LOCATABLE_DATA_LINE_TO_RECORD_FUNCTION, SIMPLE_LOCATABLE_RECORD_AND_DATA_LINE_BI_CONSUMER);
+        }
+
+        private SimpleSampleLocatableCollection(final SampleMetadata sampleMetadata,
+                                                final List<SimpleLocatable> simpleLocatables) {
+            super(sampleMetadata, simpleLocatables, SimpleLocatableTableColumn.COLUMNS, SIMPLE_LOCATABLE_DATA_LINE_TO_RECORD_FUNCTION, SIMPLE_LOCATABLE_RECORD_AND_DATA_LINE_BI_CONSUMER);
+        }
+    }
+
+    private static final SimpleSampleLocatableCollection SIMPLE_LOCATABLE_COLLECTION_EXPECTED = new SimpleSampleLocatableCollection(
+            new SimpleSampleMetadata(SAMPLE_NAME_EXPECTED),
+            Arrays.asList(
+                    new SimpleLocatable(new SimpleInterval("1", 1, 1), 1),
+                    new SimpleLocatable(new SimpleInterval("1", 2, 2), 2),
+                    new SimpleLocatable(new SimpleInterval("10", 1, 1), 3),
+                    new SimpleLocatable(new SimpleInterval("2", 1, 1), 4)));
+
+    @Test
+    public void testRead() {
+        final SimpleSampleLocatableCollection simpleLocatableCollection = new SimpleSampleLocatableCollection(SIMPLE_LOCATABLE_COLLECTION_FILE);
+        assertSimpleLocatableCollectionEqualsExpected(simpleLocatableCollection);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testReadIntervalsNotInLexicographicalOrder() {
+        new SimpleSampleLocatableCollection(SIMPLE_LOCATABLE_COLLECTION_NON_LEXICOGRAPHICAL_ORDER_FILE);
+    }
+
```

Comment:
You could add extra tests to check that the right exceptions are thrown for when a file has wrong column names, or column names are in the wrong order.

Reply:
TableWriter doesn't fail as long as the mandatory columns are there in any order.  (I do have a test for missing columns here, but really I think this should be the responsibility of the TableWriter tests.)